### PR TITLE
Add inverse projection matrix to fragment shader globals

### LINKED ISSUE #59302: Using INV_PROJECTION_MATRIX in gdshader causes editor to 100% cpu and hang on or after compilation.
### Godot version

4.0 dev 35eec4d

### System information

popOS! (Ubuntu), RTX2060, Vulkan

### Issue description

Editor uses 100% cpu and hangs on compilation of a shader. This happens as soon as the builtin INV_PROJECTION_MATRIX is added to the code. The only useful logs i could get which may or may not be related is:

   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:280)
ERROR: Error compiling Fragment shader, variant #7 (
#define MODE_MULTIPLE_RENDER_TARGETS
).
   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:276)
ERROR: Failed parse:
WARNING: 0:72: '' : all default precisions are highp; use precision statements to quiet warning, e.g.:
         "precision mediump int; precision highp float;" 
ERROR: 0:2054: 'inv_projection_matrix' : undeclared identifier 
ERROR: 0:2054: '' : compilation terminated 
ERROR: 2 compilation errors.  No code generated.

   at: _compile_variant (servers/rendering/renderer_rd/shader_rd.cpp:277)
ERROR: code:

### Steps to reproduce

create new project, add new shader resource and start writing - using INV_PROJECTION_MATRIX.

tested with:

void fragment(){
	float depth = texture(DEPTH_TEXTURE, SCREEN_UV).x;
	vec3 ndc = vec3(SCREEN_UV * 2.0 - 1.0, depth);
	vec4 view = INV_PROJECTION_MATRIX * vec4(ndc, 1.0);
}

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by nevarek**
Fixes #59302

### CODE REVIEW

File: servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl [hunk 1]
```diff
@@ -455,10 +455,12 @@ layout(location = 9) in flat uint instance_index_interp;
 
 #define model_matrix instances.data[instance_index].transform
 #ifdef USE_MULTIVIEW
-#define projection_matrix scene_data.projection_matrix_view[ViewIndex]
+	mat4 projection_matrix = scene_data.projection_matrix_view[ViewIndex];
+	mat4 inv_projection_matrix = scene_data.inv_projection_matrix_view[ViewIndex];
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl:459]: I would stick with the old way of doing it e.g: 
```suggestion
#define projection_matrix scene_data.projection_matrix_view[ViewIndex]
#define inv_projection_matrix scene_data.inv_projection_matrix_view[ViewIndex]
```

File: servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl [hunk 2]
```diff
@@ -455,10 +455,12 @@ layout(location = 9) in flat uint instance_index_interp;
...
 #else
-#define projection_matrix scene_data.projection_matrix
-#endif
+	mat4 projection_matrix = scene_data.projection_matrix;
+	mat4 inv_projection_matrix = scene_data.inv_projection_matrix;
```
Comment (clayjohn) [inline servers/rendering/renderer_rd/shaders/scene_forward_clustered.glsl:462]:
```suggestion
#define projection_matrix scene_data.projection_matrix
#define inv_projection_matrix scene_data.inv_projection_matrix
```

**Review by clayjohn — CHANGES_REQUESTED** [review]
I think it makes more sense to stick with the condition preprocessor defines for the fragment shader as the projection matrices are read only in the fragment function.

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**